### PR TITLE
Add basic ipfs pubsub messaging to app.

### PR DESCRIPTION
**Pull request by eolszewski**

### CODE REVIEW

File: packages/transmute-dashboard/src/components/Messages/Messages.js [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+});
+
+const ipfsOptions = {
+  EXPERIMENTAL: {
+    pubsub: true
+  },
+  config: {
+    Addresses: {
+      Swarm: [
+        "/dns4/ws-star.discovery.libp2p.io/tcp/443/wss/p2p-websocket-star"
```
Comment (OR13) [inline packages/transmute-dashboard/src/components/Messages/Messages.js]: this config object should probably be imported and not embedded in a component.

Comment (eolszewski) [reply packages/transmute-dashboard/src/components/Messages/Messages.js]: Agree, good catch.

File: packages/transmute-dashboard/src/components/Messages/Messages.js [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+    this.updatePeerStatus = this.updatePeerStatus.bind(this);
+  }
+
+  async componentWillMount() {
+    const user = await this.props.auth.getUser();
+    this.ipfs.once('ready', () => this.ipfs.id((err, info) => {
+      if (err) { throw err }
+      this.setState({ info });
+      
+      const map = {
```
Comment (OR13) [inline packages/transmute-dashboard/src/components/Messages/Messages.js]: map is not a safe name.

Comment (eolszewski) [reply packages/transmute-dashboard/src/components/Messages/Messages.js]: Reasonable

File: packages/transmute-dashboard/src/components/Messages/Messages.js [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+        this.room.sendTo(peer, JSON.stringify(map));
+      });
+
+      this.room.on('peer left', (peer) => {
+        this.updatePeerStatus(peer, false);
+      });
+
+      this.room.on('message', (message) => {
+        // Check if this is an introductory message
+        if (this.isMessageJSON(message.data.toString()) && _.difference(['name', 'pk', 'id', 'online'], _.keys(JSON.parse(message.data.toString()))).length === 0) {
```
[the remark this answers is not in the document]
Comment (eolszewski) [reply packages/transmute-dashboard/src/components/Messages/Messages.js]: Sure

File: packages/transmute-dashboard/src/components/Messages/Messages.js [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+  async componentWillMount() {
+    const user = await this.props.auth.getUser();
+    this.ipfs.once('ready', () => this.ipfs.id((err, info) => {
+      if (err) { throw err }
+      this.setState({ info });
+      
+      const map = {
+        id: info.id,
+        name: user.name,
+        pk: JSON.parse(user.did_document).id.split(':')[2],
```
Comment (OR13) [inline packages/transmute-dashboard/src/components/Messages/Messages.js]: This is not always a public key, sometimes its a contract address. For true DID compatibility, we should be pulling from the publicKeys array, the key in hex, and the key type.

Comment (OR13) [reply packages/transmute-dashboard/src/components/Messages/Messages.js]: I think its ok for now though

Comment (eolszewski) [reply packages/transmute-dashboard/src/components/Messages/Messages.js]: I'll update with OpenPGP and using the pem.

File: packages/transmute-dashboard/src/components/Messages/Messages.js [hunk 5]
```diff
@@ -0,0 +1,238 @@
...
+  }
+
+  async componentWillMount() {
+    const user = await this.props.auth.getUser();
+    this.ipfs.once('ready', () => this.ipfs.id((err, info) => {
+      if (err) { throw err }
+      this.setState({ info });
+
+      const secPub = openpgp.key.readArmored(JSON.parse(user.did_document).publicKey[0].publicKeyPem).keys[0];
+      const publicKey = Buffer.from(secPub.primaryKey.params[1].data).toString('hex');
```
Comment (OR13) [inline packages/transmute-dashboard/src/components/Messages/Messages.js]: much better.

**Pull request merged**